doc/classes/OS.xml

```diff
@@ -701,6 +701,24 @@
 				[b]Note:[/b] This method is implemented on Linux, macOS, and Windows.
 			</description>
 		</method>
+		<method name="precise_delay_msec" qualifiers="const">
+			<return type="void" />
+			<param index="0" name="msec" type="int" />
+			<description>
+				Delays execution of the current thread by [param msec] milliseconds with higher precision. [param msec] must be greater than or equal to [code]0[/code]. Otherwise, [method delay_msec] does nothing and prints an error message.
+				[b]Note:[/b] [method delay_msec] is a [i]blocking[/i] way to delay code execution. To delay code execution in a non-blocking way, you may use [method SceneTree.create_timer]. Awaiting with [SceneTreeTimer] delays the execution of code placed below the [code]await[/code] without affecting the rest of the project (or editor, for [EditorPlugin]s and [EditorScript]s).
+				[b]Note:[/b] When [method delay_msec] is called on the main thread, it will freeze the project and will prevent it from redrawing and registering input until the delay has passed. When using [method delay_msec] as part of an [EditorPlugin] or [EditorScript], it will freeze the editor but won't freeze the project if it is currently running (since the project is an independent child process).
```

Comment:
```suggestion
				Delays execution of the current thread by [param msec] milliseconds with higher precision. [param msec] must be greater than or equal to [code]0[/code]. Otherwise, [method precise_delay_msec] does nothing and prints an error message.
				Compared to [method delay_msec], [method precise_delay_msec] is more precise at low sleep intervals at the cost of higher CPU utilization. [method precise_delay_usec] should be preferred in scenarios where stability over time is preferred, e.g. for frame pacing.
				[b]Note:[/b] [method precise_delay_msec] is a [i]blocking[/i] way to delay code execution. To delay code execution in a non-blocking way, you may use [method SceneTree.create_timer]. Awaiting with [SceneTreeTimer] delays the execution of code placed below the [code]await[/code] without affecting the rest of the project (or editor, for [EditorPlugin]s and [EditorScript]s).
				[b]Note:[/b] When [method precise_delay_msec] is called on the main thread, it will freeze the project and will prevent it from redrawing and registering input until the delay has passed. When using [method precise_delay_msec] as part of an [EditorPlugin] or [EditorScript], it will freeze the editor but won't freeze the project if it is currently running (since the project is an independent child process).
```